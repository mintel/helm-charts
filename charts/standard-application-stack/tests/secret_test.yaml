suite: Test ExternalSecret Creation
templates:
- secrets.yaml
release:
  namespace: test-namespace
tests:
- it: ArgoCD annotations
  set:
    global.clusterEnv: test
    externalSecret:
      enabled: true
      argo:
        hook: PreSync
        hookDeletePolicy: HookSucceeded
        syncWave: "-1"
  asserts:
  - matchSnapshot: {}
  - hasDocuments:
      count: 1
  - equal:
      path: metadata.annotations["argocd.argoproj.io/hook"]
      value: PreSync
  - equal:
      path: metadata.annotations["argocd.argoproj.io/hook-delete-policy"]
      value: HookSucceeded
  - equal:
      path: metadata.annotations["argocd.argoproj.io/sync-wave"]
      value: "-1"
