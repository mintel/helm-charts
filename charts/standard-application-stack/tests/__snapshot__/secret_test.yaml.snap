ensures ArgoCD annotations are populated:
  1: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      annotations:
        argocd.argoproj.io/hook: PreSync
        argocd.argoproj.io/hook-delete-policy: HookSucceeded
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "-1"
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/env: test
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app-app
      namespace: test-namespace
    spec:
      dataFrom:
        - extract:
            key: test-namespace/example-app/app
      refreshInterval: 5m0s
      secretStoreRef:
        kind: SecretStore
        name: aws-secrets-manager-default
      target:
        creationPolicy: Owner
